models/concerns/validate_data.rb

```diff
@@ -414,6 +415,18 @@ def check_relations(periodic_heuristic)
     check_relation_visits_number_lapse_consistency
   end
 
+  def check_maximum_lateness(periodic_heuristic)
+    return nil unless periodic_heuristic
+
+    if @hash[:services].any?{ |service|
+      (service[:activity].to_a + service[:activities].to_a).any?{ |activity|
```

Comment:
`service[:activity].to_a` will transform the activity hash into an array of arrays. Instead of having a hash within an array
```suggestion
      ([service[:activity]] + service[:activities].to_a).compact.any?{ |activity|
```